halo/lua/lua.cpp

```diff
@@ -4,36 +4,88 @@
  * This program is free software under the GNU General Public License v3.0 or later. See LICENSE for more information.
  */
 
-#include "lua.hpp"
-#include "../memory/types.hpp"
-#include "../memory/global.hpp"
+#include <string>
 
-#define WIN32_MEAN_AND_LEAN
-#include <windows.h>
-#include <cstdio>
+#include <lua.hpp>
 
+#include "../../includes/file_helpers.hpp"
+#include "../hooks/map.hpp"
+#include "../memory/global.hpp"
+#include "../memory/types.hpp"
 #include "../paths.hpp"
 
+#include "lua_helpers.hpp"
+#include "lua_messaging.hpp"
+#include "lua_open.hpp"
+#include "lua.hpp"
+
+static lua_State* map_state = NULL;
```

Comment:
Is `lua_State` your thing? It's odd for a datatype to have a name like that. I'd expect it to be like `LUA_State` or `LuaState`.

Reply:
lua_State is the type in the lua library.